### DISCUSSION TITLE
3D model in app

### DISCUSSION THREAD

**Discussion by NanoscaleSimulations** [reflex-dev/reflex#688] [Q&A]
Hi all,

I had an idea for a project where the user can input some information and then see a 3D model (generated by a python library) accordingly. Is there a way to approach this using pynecone? I am aware there is something called Three JS, or is there another way to do this?

Kind regards

**Comment by gsal** [comment]
I would like to do the same, some time in the future; so, I looked and looked and found [cadquery](https://cadquery.readthedocs.io/en/latest/).  I have not used it, though, but went through the docs...looks good.

**Reply by NanoscaleSimulations** [reply]
Looks interesting, but how can it be used in combination with pynecone? Will a pynecone app be able to render a 3D model using this?

**Comment by gsal** [comment]
Don't know for sure; I don't know Pynecone well at all, I am just getting started.

cadquery can export a variaty of [formats](https://cadquery.readthedocs.io/en/latest/importexport.html), including ThreeJS which is just JavaScript and is supported by browsers right out of the box.  I don't know what would be needed, maybe some generic graphic context (3D?) canvas or something.  Push come to shove, maybe there is need to wrap some react component or something?

cadquery can also export in VRML format, although not supported right out of the box by browsers, there are plug-ins for this.

Maybe someone from the Pynecone team can comment on this and help us out.

If they need a ThreeJS-based 3D model to test with, maybe we can provide one from the cadquery site (or generate one from the examples) and maybe, just maybe, the pynecone team can provide us with a recipe on how to render a ThreeJS file.

**Comment by tgberkeley** [comment]
Hey @gsal @NanoscaleSimulations,
So I believe that you can do this with Spline (https://spline.design). If you export the model and then wrap it with react then you should be able use it in your app. Here is a link to some code that might help: https://github.com/reflex-dev/reflex-web/blob/5b513d3d584f0a2b86aa3362f1d952b66988a9bb/pcweb/components/chat.py#L5-L29
